src/content/blog/val-vibes.mdx

```diff
@@ -0,0 +1,141 @@
+---
+title: "Val Vibes: Semantic search in Val Town"
+description: How to build semantic search for Val Town within Val Town itself
+pubDate: June 18, 2024
+author: JP Posma
+---
+
+import { Image } from "astro:assets";
+import ManBitesDog from "./val-vibes/man-bites-dog.png";
+import CompareEmbeddings from "./val-vibes/compare-embeddings.png";
+import QueryVals from "./val-vibes/query-vals.png";
+import Indexing1 from "./val-vibes/indexing1.png";
+import Indexing2 from "./val-vibes/indexing2.png";
+import Indexing3 from "./val-vibes/indexing3.png";
+import Search1 from "./val-vibes/search1.png";
+import Search2 from "./val-vibes/search2.png";
+import Blobindex from "./val-vibes/blobindex.png";
+import Blobsearch from "./val-vibes/blobsearch.png";
+import ManLicksDog from "./val-vibes/man-licks-dog.jpg";
+
+### Similar vibes
+
+“Search” is finding documents based on a query, using some sort of sorting mechanism that ranks the most relevant documents. One such sorting mechanism is “semantic search”, which uses one of those crazy features to come out of the world of machine learning: embeddings. It works like this: you put in a string into an API, and you get an array of numbers (the “embedding”), with the property that things that have the same “vibe” will get similar numbers.
+
+For example, “animal that barks” should have similar vibes to “dog”. And “dog bites man” should have different vibes than “man bites dog”.
+
+<figure style={{ maxWidth: "300px" }}>
+  <Image src={ManBitesDog} alt="Man bites dog" />
+  <figcaption>
+  </figcaption>
+</figure>
+
+You generate embeddings for all your documents (ahead of time), as well as for the search query, and then you return documents sorted by how similar the embeddings are.
+
+That's the idea, though in practice embeddings can be quite unpredictable and surprising. Our examples above already break down… “animal that barks” apparently vibes more with “woof woof woof” than with “dog”. And “dog bites man” and “man bites dog” have even more similar vibes than that, presumably because there's lots of biting going on in both cases.
```

Comment:
In some sense I love this example but maybe it is more muddling than helpful. I think for val town the search terms are more like "custom gpt" or "discord bot" or "sendgrid attachment" where we'll get a whole lot of results near 0 similarity and then a couple closer to 0.5. Basically for our use-case we mostly want to surface anything at all similar and we don't care as much about disambiguating who is biting whom